src/settings.rs

```diff
@@ -0,0 +1,138 @@
+use glib::{GString, Variant};
+use SettingsExt;
+
+pub trait SettingsGetValueType {
+    fn get<T: SettingsExt>(&T, key: &str) -> Self;
+}
+
+pub trait SettingsSetValueType {
+    fn set<T: SettingsExt>(&T, key: &str, val: Self) -> bool;
+}
```

Comment:
I've split this into two Traits, since sometimes we return a different value in `set_` than we take in `get_` (e.g. with `get/set_string`, we take `&str` in `set_string` but return `Option<GString>` in `get_string`